app/lib/argocd_client.rb

```diff
@@ -17,4 +17,61 @@ def initialize
     @helm_chart_name = Figaro.env.HELM_CHART_NAME
     @helm_chart_version = Figaro.env.HELM_CHART_VERSION
     @helm_chart_repository = Figaro.env.HELM_CHART_REPOSITORY
+    @conn = Faraday.new(
+      url: @url,
+      headers: {
+          "Content-Type" => "application/json",
+          "Authorization": "Bearer " + @token
+      }
+    )
   end
+
+  def get_cluster_map()
+    puts("in get_cluster map")
+    cluster_map = JSON.parse(@conn.get('/api/v1/clusters').body)['items'].select { |i| i['connectionState']['status'] == 'Successful' }.map { |i| [i['name'], i['server']] }.to_h
+
+    cluster_map
+  end
+
+  def create_application(app_group_name, override_values)
+    output = @conn.post do | req |
+      req.body = {
+        metadata: {
+          name: "#{Figaro.env.argocd_project_name}-#{app_group_name}-#{Figaro.env.argocd_default_destination_name}",
+          namespace: Figaro.env.argocd_namespace
+        },
+        spec: {
+          destination: {
+            server: Figaro.env.argocd_default_destination_server,
+            namespace: Figaro.env.argocd_destination_namespace
+          },
+          project: Figaro.env.argocd_project_name,
+          source: {
+            repoURL: Figaro.env.helm_repository,
+            chart: Figaro.env.helm_chart_name,
+            targetRevision: Figaro.env.helm_chart_version,
+            helm: {
+              releaseName: app_group_name,
+              values: YAML.dump(override_values),
+            }
+          }
+        }
+      }.to_json
+      req.params['upsert'] = true
+      req.path = '/api/v1/applications'
+    end
+    output
+
+  end
+
+  def sync_application(app_group_name)
+    output = @conn.post do | req |
+      req.body = {
+        project: Figaro.env.argocd_project_name,
+      }.to_json
+      req.path = "/api/v1/applications/#{Figaro.env.argocd_project_name}-#{app_group_name}-#{Figaro.env.argocd_default_destination_name}/sync"
```

Comment:
`Figaro.env.argocd_default_destination_name` is only temporary, we should receive it from the models that @bentol creates instead of hardcode it like this I guess

Reply:
got it.